### DISCUSSION TITLE
Unmounted input not validating

### DISCUSSION THREAD

**Discussion by RamsayRomero** [react-hook-form/react-hook-form#9013] [Q&A]
Here's a simple example of the issue I'm having: [https://codesandbox.io/s/eager-sky-obqnqf?file=/src/App.js](https://codesandbox.io/s/eager-sky-obqnqf?file=/src/App.js)

Basically I have a list of components with inputs and I'd only like to render one of them at a time based on the current selected state. I can switch between these components and edit their values and when I submit the form, both values are logged correctly even though only one input is mounted. However, the validation is not triggered for the unmounted component. If you clear the input's value, and then select the other input, the form still gets submitted even though both values are required.

Any solutions for triggering validation for unmounted inputs, suggestions for different ways to architect the form altogether, or understanding for why the values are being tracked but not the validation rules would be greatly appreciated :). I know I could do something like render all the list items and only show the selected one and hide the others, but I would prefer to only render one at a time as my actual use case involves potentially thousands of components to select from. Thanks!

**Comment by leapful** [comment]
@RamsayRomero : Native validation of react-hook-form is skipped if input element is not mounted. You should use custom resolver like yup to perform validation.

https://github.com/react-hook-form/react-hook-form/blob/104314d38b9c3c9270fc11ca217169ef01871309/src/logic/validateField.ts#L49-L51

Working demo: https://codesandbox.io/s/eloquent-snow-sqhmrf?file=/src/App.js
